auraed/src/cells/cell_service/cells/cell.rs

```diff
@@ -234,12 +234,9 @@ impl CellsCache for Cell {
         children.broadcast_kill()
     }
 
-    fn cell_graph(&self, node: GraphNode) -> GraphNode {
-        let CellState::Allocated { children, .. } = &self.state else {
-            return GraphNode {
-                cell_info: Some((self.cell_name.clone(), self.spec.clone())),
-                children: vec!(),
-            };
+    fn cell_graph(&mut self, node: GraphNode) -> GraphNode {
+        let CellState::Allocated { children, .. } = &mut self.state else {
+            return GraphNode::Invalid { cell_name: self.cell_name.clone() };
```

Comment:
Looks like you are trying to create an error case. You should have the function's return be a `Result<GraphNode, CellsError>` instead. See the functions above.

Reply:
The thing is I actually need the `CellName` to be able to purge it from the cache, that's why I introduced this enum.

Reply:
https://github.com/JeroenSoeters/aurae/blob/list-cells/auraed/src/cells/cell_service/cells/cells.rs#L250-L265

Reply:
CellsError::CellNotAllocated has a `cell_name` field

Reply:
Refactored it again to a `Result` type and it looks a lot cleaner now. Thanks!